### DISCUSSION TITLE
Order of outputs of ConfusionMatrix

### DISCUSSION THREAD

**Discussion by ashleve** [Lightning-AI/torchmetrics#287] [Q&A] [answered]
Hi,
is there any explanation on order of output generated by ConfusionMatrix?

Here is the example of binary classification from the docs:
```python
>>> from torchmetrics import ConfusionMatrix
>>> target = torch.tensor([1, 1, 0, 0])
>>> preds = torch.tensor([0, 1, 0, 0])
>>> confmat = ConfusionMatrix(num_classes=2)
>>> confmat(preds, target)
tensor([[2., 0.],
        [1., 1.]])
```

Judging by the output thensor, I assume the order in matrix is:
```
[ True Negative, False Negative ]
[ False Positive, True Positive ]
```
Did I get it right?
Why is the order like that exactly?
For example, on wikipedia it looks like this:
![image](https://user-images.githubusercontent.com/43458894/120858212-c6672180-c582-11eb-8a90-7b66123d6cdb.png)
I suggest to add some explanation to the docs about order and interpretation of the outputs, becuase currently this is confusing.

**Comment by justusschock** [comment, marked as answer]
Hi this is actually the same as wikipedia gives just not with positive and negative but with an arbitrary number of classes. I.e. You have predicted class on the x axis and actual class on the y axis. Both start with class zero and the diagonal is always the number of correct predictions for the corresponding class
